### PR TITLE
chore(repo): add fetch functions

### PR DESCRIPTION
**Pull request by samuelmasse**
We had way too many not nulls everywhere in the code, so this PR seperates all our `get` functions into `get` and `fetch`. A fetch functions is going to try to get the data, and return undefined if it doesn't find it. The get function will use the fetch function, and if the fetch function returns undefined, it will throw an error. This way we can use the get function in places in the code where we expect the get function to return a value, and remove all our not nulls. If for any reason the get function fails, it returns a clear message and prevents the request from running any longer with garbage data.

Closes DEV-2166

### CODE REVIEW

File: packages/server/src/clients/service.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class ClientService extends Service {
     return client
   }
 
-  async getById(id: uuid): Promise<Client | undefined> {
+  async fetchById(id: uuid): Promise<Client | undefined> {
```
Comment (laurentlp) [inline packages/server/src/clients/service.ts:59]: Private?

File: packages/server/src/messages/api.ts [hunk 2]
```diff
@@ -152,12 +152,12 @@ export class MessageApi {
...
 
-    const message = await this.messages.get(id)
+    const message = await this.messages.fetch(id)
     if (!message) {
       return res.sendStatus(404)
     }
 
-    const conversation = await this.conversations.get(message.conversationId)
+    const conversation = await this.conversations.fetch(message.conversationId)
     if (!conversation || conversation.clientId !== req.client!.id) {
```
Comment (laurentlp) [inline packages/server/src/messages/api.ts:161]:
```suggestion
    if (!conversation || conversation.clientId !== req.client.id) {
```

**Review by laurentlp — APPROVED** [review]

File: packages/server/src/messages/api.ts (hunk 2 above)
Comment (samuelmasse) [reply packages/server/src/messages/api.ts]: this is fixed

File: packages/server/src/clients/service.ts (hunk 1 above)
Comment (samuelmasse) [reply packages/server/src/clients/service.ts:59]: The fetch function can and is often used outside the service. It's not a public/private thing. The difference between the get and fetch function is intent. With the get function you indicate that the result must not be null, whereas with the fetch you indicate that you will handle the null value yourself.

**Pull request merged**